src/cogent3/core/new_alignment.py

```diff
@@ -356,35 +395,72 @@ def take_seqs(
         negate
             select all sequences EXCEPT names
         kwargs
-            keyword arguments to be passed to the constructor of the new seqcollection
+            keyword arguments to be passed to the constructor of the new collection
+
+        Notes
+        -----
+        The seqs in the new collection will be references to the same objects as
+        the seqs in the old collection.
         """
+        # todo: kath, note gavins comment that this method could operate on only the names
+        # list and not the seqs_data object
 
-        # todo: kath, add subset method onto SeqsData
         if negate:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in self.names
-                if name not in names
-            }
+            names = [name for name in self.names if name not in names]
         else:
-            selected_seqs = {
-                name: self.seqs.get_seq_str(seqid=name)
-                for name in names
-                if name in self.names
-            }
+            names = [name for name in names if name in self.names]
 
-        if not selected_seqs:
+        if not names:
             return {}
```

Comment:
not a good return value. Better to raise an exception, like `ValueError(f"{names=} and {negate=} resulted in no names")`